tests/brevitas/test_onnx_export.py

```diff
@@ -145,9 +146,27 @@ def test_dynamic_quantization(
                 onnx_config_class_constructor=onnx_config_class_constructor,
                 shapes_to_validate=VALIDATE_EXPORT_ON_SHAPES,
             )
+            original_matmul_gemm_counter = 0
+            onnx_model = onnx.load(os.path.join(tmpdir, "model.onnx"))
+
+            for node in onnx_model.graph.node:
+                if node.op_type == "Gemm" or node.op_type == "MatMul":
+                    original_matmul_gemm_counter += 1
 
+            find_and_insert_matmulinteger(tmpdir)
             onnx_model = onnx.load(os.path.join(tmpdir, "model.onnx"))
 
+            matmul_gemm_counter = 0
+            matmulinteger_counter = 0
             for node in onnx_model.graph.node:
-                # Check that we have MatmulInteger, etc.
-                pass
+                if node.op_type == "Gemm" or node.op_type == "MatMul":
+                    matmul_gemm_counter += 1
+
+                if node.op_type == "MatMulInteger":
+                    matmulinteger_counter += 1
+
+            # The number of Matmul+Gemm has to be less compared to the model pre-transformation
+            # This is not zero since there are matmul that are not linear layers so they are not replaced
+            # and some linears layers can be excluded from quantization
+            assert matmul_gemm_counter <= original_matmul_gemm_counter
```

Comment:
A better test would have us monitoring exactly how many Matmul and Gemm we expect to have before/after transformations, and similarly with matmulinteger.

Considering all the other changes that we are doing plus new tests that we will be adding, maybe we could wait for this kind of implementation when everything is more stable.